plugin/metric/collector/instance.go

```diff
@@ -0,0 +1,51 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/store"
+	"go.uber.org/zap"
+)
+
+// instanceCollector is the metric data collector for instance.
+type instanceCollector struct {
+	l     *zap.Logger
+	store *store.Store
+}
+
+// instanceEventName is the MetricEventName for instance
+var instanceEventName MetricEventName = "bb.instance"
+
+// NewInstanceCollector creates a new instance of instanceCollector
+func NewInstanceCollector(l *zap.Logger, store *store.Store) MetricCollector {
+	return &instanceCollector{
+		l:     l,
+		store: store,
+	}
+}
+
+// Collect will collect the netric for instance
+func (c *instanceCollector) Collect(ctx context.Context) ([]*Metric, error) {
+	var res []*Metric
+
+	status := api.Normal
+	instanceCountMap, err := c.store.CountInstanceGroupByEngine(ctx, &api.InstanceFind{
+		RowStatus: &status,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for engine, count := range instanceCountMap {
+		res = append(res, &Metric{
+			EventName: instanceEventName,
```

Comment:
For metric normally there is a value and a bunch of labels (dimensions)

e.g. the instance count metric

The value is the instance count, while the labels are "database", "Environment" etc.

Reply:
Also I would suggest to name metric as bb.instance.count.

because we may collect metrics like bb.instance.create (record the number of instance created)

Reply:
An example https://opentelemetry.io/docs/reference/specification/metrics/semantic_conventions/process-metrics/

Reply:
OpenTelemetry is an effort to standardize log/metric/tracing, though they are not mature. FYI

Reply:
So do you mean we'd better using a slice like `[{ "label": "type", "value": "MySQL" }, { "label": "count", "value": 10 }]` instead of a map for dimensions here?

Reply:
value: 10
label : {
  engine: mysql
  environment: prod
}

Reply:
maybe:

```go
type Metric struct {
	Name       MetricName
	Value      int
	Dimensions map[string]string
}

&Metric{
  Name: "bb.instance.count",
  Value: 10,
  Dimensions: map[string]string{
    "engine": "mysql",
    "environment": "prod",
  }
}
```

Reply:
Updated, PTAL

Reply:
We may call it Label to be consistent with Cloud metrics.
https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries

Reply:
Label is a more commonly adopted name.

AWS https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-metric-streams-formats-opentelemetry.html

Google https://cloud.google.com/monitoring/api/resources

LG otherwise